Dockerfile

```diff
@@ -16,7 +16,7 @@ RUN \
   apt-get install -y --no-install-recommends unattended-upgrades python3=3.5.1-3 python3-dev=3.5.1-3 && \
   apt-get install -y --no-install-recommends python3-pip=8.1.1-2ubuntu0.4 python3-yaml=3.11-3build1 && \
   apt-get install -y --no-install-recommends language-pack-en=1:16.04+20161009 git=1:2.7.4-0ubuntu1.3 && \
-  apt-get install -y --no-install-recommends ssh=1:7.2p2-4ubuntu2.2 && \
+  apt-get install -y --no-install-recommends ssh=1:7.2p2-4ubuntu2.4 && \
```

Comment:
I don't think it's wise to pin specific versions of every apt package. Package versions get removed from apt all the time (for example if vulnerabilities are found) and it would make the image unbuildable.

Reply:
Well, specific versions are recommended by [codacy](https://app.codacy.com). More details:
- https://app.codacy.com/app/rjeczalik/koding/issues?step=20: 
   ```
   Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
   ```
- https://github.com/hadolint/hadolint/wiki/DL3008
- https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#label
  > Version pinning forces the build to retrieve a particular version regardless of what’s in the cache. This technique can also reduce failures due to unanticipated changes in required packages.

In case vulnerabilities are found I like the fact the build fails. That forces to search for newer versions and specify those, rather than go for latest version. In case you just want to pull the docker image, you should get it from the respective docker registry (zalando internal as of now).

Reply:
My main concern is that Amazon updates ssh server and ghe-backup stops working because the ssh client was not updated - and knowing ssh error messages it might not even be obvious at first. Pining this version is not even that useful since it depends on non pinned versions of openssh (https://packages.ubuntu.com/xenial/ssh).

Reply:
The pinned python3.5 versions also seem more harmful than good since the package will never update to an incompatible version of python (3.6 gets it's own package) and it prevents getting the beneficts of bug fixes and optimizations done in later patch versions. It also moves the burden of keeping a supported patch version of python to us (for example 3.5.1 doesn't include any security fix from the last 2 and half years and the current supported patch version is 3.5.5).

Reply:
thanks for the 3.5.5 hint - will include that version update in this PR